pennylane/gradients/param_shift_hessian.py

```diff
@@ -211,17 +241,24 @@ def param_shift_hessian(tape, f0=None):
 
     >>> qml.jacobian(qml.grad(cost))(weights)
 
-    .. note::
-
-        Currently, parametric gates are only supported if they obey a two-term shift rule,
-        which includes the following operations:
-
-        "RX", "RY", "RZ", "Rot", "PhaseShift", "ControlledPhaseShift", "MultiRZ", "PauliRot",
-        "U1", "U2", "U3", "SingleExcitationMinus", "SingleExcitationPlus", "DoubleExcitationMinus",
-        "DoubleExcitationPlus", "OrbitalRotation".
-
     Args:
         tape (pennylane.QNode or .QuantumTape): quantum tape or QNode to differentiate
+        argnum (int or list[int] or None): Parameter indices to differentiate
+            with respect to. If not provided, the Hessian with respect to all
+            trainable indices is returned. Note that the indices refer to tape
+            parameters if ``tape`` is a tape, and to QNode arguments if it is a QNode.
```

Comment:
Actually I think the `argnum` indices always refer to the trainable tape arguments even for qnodes, see this example:

```py
@qml.qnode(qml.device("default.qubit", wires=2))
def circuit(x):
    qml.RY(x[0], wires=0)
    qml.RX(x[1], wires=0)
    return qml.probs(0), qml.probs(1)
qml.gradients.param_shift(circuit, argnum=1)(qml.numpy.array([0.1, 0.2]))
```

Reply:
Adapted the docstring, all further edits regarding the logic should be taken care of together with `param_shift`